awx/main/scheduler/task_manager.py

```diff
@@ -572,6 +596,8 @@ def process_tasks(self, all_sorted_tasks):
         pending_tasks = [t for t in all_sorted_tasks if t.status == 'pending']
         undeped_tasks = [t for t in pending_tasks if not t.dependencies_processed]
         dependencies = self.generate_dependencies(undeped_tasks)
+        deps_of_deps = self.generate_dependencies(dependencies)
+        dependencies += deps_of_deps
```

Comment:
I remember you mentioning this, but now I get it. Basically, `deps_of_deps` can only be the project updates generated for inventory updates. Because you know it only goes this-many levels deep it's written like this and not iterated indefinitely.

Reply:
exactly